src/longsequences/copying.jl

```diff
@@ -37,10 +118,10 @@ end
         x = dstdata[index(id)]
         y = srcdata[index(is)]
         if offset(id) < offset(is)
-            y >>= offset(is) - offset(id)
+            y >>= (offset(is) - offset(id)) & 63
```

Comment:
Does removing this branching through something like the following generate faster code?

```julia
mn = min(offset(is), offset(id))
mx = max(offset(is), offset(id))
y >>= (mx - mn) & 63
k = min(64 - mx, rest)
```

Reply:
Unfortunately not. `y` needs to be able to be left-shifted, which it can't in your code snippet. It's possible to avoid the branch, but since the branch is highly predictable (it always alternates), I don't know how much performance there is to gain.
It's possible to optimize this section of the code by specializing it more - note that except for the very first iteration, either `offset(is)` or `offset(id)` is zero. Sometimes both will be zero, and it can be made *much* more efficient. I don't think it's worth the code complexity though, ATM my laptop does about 4 bits/cycle. Pretty good. If we ever see a compelling case to optimize this further, I'm sure it can be done (probably also SIMD'd, actually).